tests/containers/upload_container_images_to_ecr.pm

```diff
@@ -0,0 +1,39 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Push a container image to the public cloud container registry
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.com>, qa-c team <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use publiccloud::ecr;
+use containers::urls 'get_suse_container_urls';
+
+sub run {
+    my ($self, $args) = @_;
+    $self->select_serial_terminal;
+
+    my $provider = publiccloud::ecr->new(
+        key_id     => get_var('PUBLIC_CLOUD_KEY_ID'),
+        key_secret => get_var('PUBLIC_CLOUD_KEY_SECRET'),
+        region     => get_var('PUBLIC_CLOUD_REGION', 'eu-central-1'),
+        username   => get_var('PUBLIC_CLOUD_USER',   'ec2-user')
+    );
+
```

Comment:
```suggestion
$self->{provider} = $provider;

```
Or better use the [provider_factory()](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/basetest.pm#L23) somehow.

Reply:
About using the provider factory, the problem is that we have (EC2) it should be changed to ECR but, we'll have more that one provider, one per amazon service and for the lasts tests (not this one) we'll need two ECR + EKS, and maybe more in the future. A simple solution is to have all in the EC2 but is a mess and we created the super class for AWS to avoid that. Options:
- We can add other option to PUBLIC_CLOUD_PROVIDER with the name ECR but we have to imagine what we have to do if we need to use the EKS
- We can follow the option 1 and put ECR + EKS together in the same class, then we should change the name of ecr.pm to something like ecr-eks.pm
- The test doesn't have the publiccloud::basetest base. I don't know if this is possible without having problems
- Or we keep the additional providers approach

For the next commit I will use your suggested solution

Reply:
What about `get_var('PUBLIC_CLOUD_AWS_SERVICE', 'ec2')` and then instantiate one or the other class.

Reply:
Let's see now. It's in the new commit